src/warc2zim/replay/notFoundPage.html

```diff
@@ -0,0 +1,26 @@
+<html>
+<head>
+<script>
+  const URL = "$URL";
+  const mainUrl = "$MAIN_URL";
+
+  if (window.parent === window.top) {
+
+    // check if URL is from same domain as the main url
+    if (!URL.startsWith(mainUrl)) {
+      // check if kiwix-serve 'blocking' is enabled, show interstitional if it is
+      if (window.block_path) {
```

Comment:
Please submit a PR on kiwix-lib to add a comment on top of this so maintainers know that other tools rely on this
https://github.com/kiwix/kiwix-lib/blob/5f8c0998291d0778887373355bb6b1c5a345ed2c/static/skin/block_external.js#L1